### PR TITLE
Org branch 2

### PR DESCRIPTION
**Pull request by DorcasOJ**
- created an endpoint to delete an existing organisation and its users
- create an endpoint to get all organisations

### CODE REVIEW

File: src/organisation/organisation.service.ts [hunk 1]
```diff
@@ -157,4 +202,202 @@ export class OrganisationService {
...
+   * */
+  async checkForAdminRole(
+    id: string,
+    profileId: string,
+    error_string = 'Only admin can add user to this organisation',
+  ) {
+    /* ensure that only admin can add users to org*/
+    const user = await this.profileOrgRepo.findOne({
+      where: { profileId: profileId, organisationId: id },
+    });
```
Comment (Chimexx) [inline src/organisation/organisation.service.ts:353]: What if user.role === orgRole.ADMIN what happens?

**Review by Chimexx — CHANGES_REQUESTED** [review]
Generally you can move some of the functions in service to utility folder, the file is too long.

File: src/organisation/organisation.controller.ts [hunk 2]
```diff
@@ -47,4 +63,52 @@ export class OrganisationController {
...
+  ) {
+    const deletedOrg = await this.orgService.removeOrganisation(id, profileId);
+    return ZapiResponse.Ok(
+      deletedOrg,
+      'Organisation and organisation teammates deleted',
+      '200',
+    );
+  }
+
+  @Delete('/:profileId/deleteUser/:id/:userEmail')
```
[the remark this answers is not in the document]
Comment (DorcasOJ) [reply src/organisation/organisation.controller.ts:96]: Okay

File: src/organisation/organisation.service.ts (hunk 1 above)
Comment (DorcasOJ) [reply src/organisation/organisation.service.ts:353]: It continues.
user.role is an enum, if it's not orgRole.ADMIN then it's orgRole.DEVELOPER.

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]
There are a couple of bugs in this code asides the one i pointed out. That's why i specifically requested only short code reviews. You need the fix these ones and drop some of the endpoints you have here. 

I noticed you also had helper functions, which is good practice but the code becomes tougher to review for ur colleagues. And u end up making more mistakes

So you need to trim down the endpoints for review. ASAP @DorcasOJ

File: src/organisation/organisation.service.ts [hunk 3]
```diff
@@ -138,6 +156,12 @@ export class OrganisationService {
...
 
+  /**
+   * It find organisation by Id
+   * if id does not exist it returns an error
+   * @Param {string} id - the organisation id
+   * @returns a promise of the organisation
+   * */
   async findOrganisationById(id: string): Promise<Organisation> {
     try {
       return await this.orgRepo.findOne(id);
```
[the remark this answers is not in the document]
Comment (DorcasOJ) [reply src/organisation/organisation.service.ts:167]: the catch is in line 168-172

File: src/organisation/organisation.service.ts [hunk 4]
```diff
@@ -148,6 +172,27 @@ export class OrganisationService {
...
+    }
+  }
+
+  /**
+   * Its gets all organisation
+   * @returns it returns a promise of the all organisation
+   * */
   async getAllOrganisation(): Promise<Organisation[]> {
     try {
       return await this.orgRepo.find();
```
[the remark this answers is not in the document]
Comment (DorcasOJ) [reply src/organisation/organisation.service.ts:198]: the catch is in line 199-203

**Comment by DorcasOJ** [comment]
Two endpoints were removed
-updateOrganisation
-deleteOneUser

**Review by andyjrII — APPROVED** [review]

**Review by Chimexx — APPROVED** [review]

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**